spring-ai-core/src/main/java/org/springframework/ai/chat/client/DefaultChatClient.java

```diff
@@ -88,18 +90,44 @@ public class DefaultChatClient implements ChatClient {
 	private final DefaultChatClientRequestSpec defaultChatClientRequest;
 
 	public DefaultChatClient(DefaultChatClientRequestSpec defaultChatClientRequest) {
+		Assert.notNull(defaultChatClientRequest, "defaultChatClientRequest cannot be null");
 		this.defaultChatClientRequest = defaultChatClientRequest;
 	}
 
-	private static AdvisedRequest toAdvisedRequest(DefaultChatClientRequestSpec inputRequest, String formatParam) {
+	private static AdvisedRequest toAdvisedRequest(DefaultChatClientRequestSpec inputRequest,
+			@Nullable String formatParam) {
+		Assert.notNull(inputRequest, "inputRequest cannot be null");
+
 		Map<String, Object> advisorContext = new ConcurrentHashMap<>(inputRequest.getAdvisorParams());
 		if (StringUtils.hasText(formatParam)) {
 			advisorContext.put("formatParam", formatParam);
 		}
 
-		return new AdvisedRequest(inputRequest.chatModel, inputRequest.userText, inputRequest.systemText,
-				inputRequest.chatOptions, inputRequest.media, inputRequest.functionNames,
-				inputRequest.functionCallbacks, inputRequest.messages, inputRequest.userParams,
+		// Process userText, media and messages before creating the AdvisedRequest.
```

Comment:
This piece of logic used to be executed only in one possible branch of execution (when using `.prompt(Prompt prompt)`). I moved it here so to ensure consistent processing of the messages before creating the `AdvisedRequest`, no matter how we get here (for example, when using `.messages(List<Message> messages)`, which was previously behaving inconsistently).

Reply:
yeh, good catch. 
Initially the ChatClinet from Prompt used to have a very different path, excluded from the advisors. 
I've tried to unify it with there unsealed path (by unsealing the prompt) but have missed the ordering of this section.